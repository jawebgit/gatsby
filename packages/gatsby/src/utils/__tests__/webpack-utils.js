const utils = require(`../webpack-utils`)

let config
beforeAll(async () => {
  config = await utils({
    stage: `develop`,
    program: {
      browserslist: [],
    },
  })
})

describe(`webpack utils`, () => {
  describe(`js`, () => {
    it(`adds .js rule`, () => {
      expect(config.rules.js).toEqual(expect.any(Function))
    })

    it(`returns default values without any options`, () => {
      const rule = config.rules.js()

      expect(rule).toMatchSnapshot()
    })

    it(`adds dependency rule`, () => {
      expect(config.rules.dependencies).toEqual(expect.any(Function))
    })

    it(`returns default values without any options`, () => {
      const rule = config.rules.dependencies()

      expect(rule).toMatchSnapshot()
    })
  })
})
