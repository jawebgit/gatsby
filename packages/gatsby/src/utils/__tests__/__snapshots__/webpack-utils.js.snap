// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack utils js returns default values without any options 1`] = `
Object {
  "exclude": /\\(node_modules\\|bower_components\\)/,
  "test": /\\\\\\.\\(js\\|mjs\\|jsx\\)\\$/,
  "type": "javascript/auto",
  "use": Array [
    Object {
      "loader": "<PROJECT_ROOT>/packages/gatsby/src/utils/babel-loader.js",
      "options": Object {},
    },
  ],
}
`;

exports[`webpack utils js returns default values without any options 2`] = `
Object {
  "exclude": /@babel\\(\\?:\\\\/\\|\\\\\\\\\\{1,2\\}\\)runtime\\|core-js/,
  "test": /\\\\\\.\\(js\\|mjs\\)\\$/,
  "type": "javascript/auto",
  "use": Array [
    Object {
      "loader": "<PROJECT_ROOT>/packages/gatsby/src/utils/babel-loader.js",
      "options": Object {
        "babelrc": false,
        "compact": false,
        "configFile": false,
        "presets": Array [
          Array [
            "<PROJECT_ROOT>/packages/babel-preset-gatsby/dependencies.js",
            Object {
              "stage": "develop",
            },
          ],
        ],
        "sourceMaps": false,
      },
    },
  ],
}
`;
